orthoinference/src/main/java/org/reactome/orthoinference/PathwayDiagramGenerator.java

```diff
@@ -0,0 +1,71 @@
+package org.reactome.orthoinference;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.pathwaylayout.PredictedPathwayDiagramGeneratorFromDB;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.Collection;
+
+public class PathwayDiagramGenerator {
+
+    private static final Logger logger = LogManager.getLogger();
+
+    private static MySQLAdaptor dba;
+    private static GKInstance speciesInst;
```

Comment:
Is it necessary for speciesInst, personId, and referenceSpeciesId to be declared as static? I am not sure I understand why they are declared like this.

Reply:
Because the static concept is still a weird one to me -- I'll remove them

Reply:
You can read more here: https://docs.oracle.com/javase/tutorial/java/javaOO/classvars.html
"static" is not a great word for the concept, I think.

A static class member will be shared by all instances of that class. For some things, such as  variables that access common resources (log files, databases, large caches of stuff), declaring them as static is a Good Idea. The variables speciesInst, personId, and referenceSpeciesId  could _potentially_ be different for each instance. So they probably should _not_ be static. 

One way to decide if something should be static is to ask: If two instances of _this_ class share the variable (if one instance modifies it, all others will see that modification), is that going to be a problem?

In this case, if you created multiple `PathwayDiagramGenerator` objects, the current code would force them to all share the values of speciesInst, personId, and referenceSpeciesId. This might be a problem if you wanted each on to create diagrams for different species, or using different reference species. The personId will _probably_ always be the same for all of them, but my personal rule is I usually do _not_ make things static, unless there is a good reason to, such as a shared database connection, or logger.

Reply:
Something about the way I've written the code requires these instances to be static. I get the message 'non-static field cannot be referenced from a static context'. I think I might need to refactor a bit in order to get around this. We did something similar for StableIdentifierGenerator, and I don't think it was too much work. I'll look into doing that since I think we want to move towards that style

Reply:
> non-static field cannot be referenced from a static context

...means that you are trying to use an _instance_ member (declared without the "Static" modifier) in a _static_ method (the "static context" that the message is referring to). The method `generateOrthologousPathwayDiagrams` is a static method, and that is probably causing the problem. 

So the question is: why is `generateOrthologousPathwayDiagrams` a static method? A static method operates outside the context of a specific instance. In fact, a static method in Java is probably better referred to as a "function", since it does not operate in the context of any instance of its class.

In this specific case here, it probably makes sense to _remove_ the static modifier from `generateOrthologousPathwayDiagrams` and make it into a proper instance method. Removing the static modifier from personId, speciesInst, and referenceSpeciesInst also makes sense. Then you can create _instances_ of `PathwayDiagramGenerator` that generate diagrams for different species. Each object will then have its _own_ internal state describing which species, reference species, and person to use when creating diagrams.

An older article, but well-written:
https://javarevisited.blogspot.com/2013/07/when-to-make-method-static-in-java.html

See also this discussion, on when to use static for methods: https://stackoverflow.com/questions/2671496/java-when-to-use-static-methods

Reply:
Thanks I'll look at both these articles and then I'll refactor a bit to make the method non-static

Reply:
I've converted the method so that it is non-static now